### PR TITLE
fix(ingest-space): route alkemio-knowledge-base BoKs to lookup.knowledgeBase() (spec 033)

### PR DESCRIPTION
**Pull request by valentinyanakiev**
## Summary

Combined PR for the ingest-space knowledge-base routing fix **and** its SDD spec (033). Fix code is identical to PR #98 (commit `0a48620`); this PR adds the spec artifacts on top.

- `plugins/ingest_space/space_reader.py` — adds `BOK_TYPE_*` constants, `KNOWLEDGE_BASE_QUERY`, `read_knowledge_base_tree()`, and the `read_body_of_knowledge()` dispatcher. `_process_space` takes an optional `top_doc_type` kwarg so the KB root document is tagged `DocumentType.KNOWLEDGE` (existing space path unchanged).
- `plugins/ingest_space/plugin.py` — routes through `read_body_of_knowledge(event.type)` and logs the resolved BoK type at handle-time.
- `tests/plugins/test_ingest_space.py` — `TestKnowledgeBaseReader` (4 tests, including a GraphQL guardrail that asserts the query string contains `knowledgeBase(ID:`), `TestBodyOfKnowledgeDispatcher` (3 tests covering both known types + unknown-type fallback), `TestIngestSpacePluginDispatchesOnType` (2 plugin-level dispatch tests).
- `specs/033-ingest-space-kb-routing/` — full SDD artifact set: spec, plan, research, data model, GraphQL contract, quickstart, tasks (all `[X]`), and quality checklist. Two remediation commits applied after `/speckit.analyze` (A5 + A6).

## Impact

On acceptance, 69 / 238 VCs (~29 %) are backed by `alkemio-knowledge-base`. Before this fix, every one of them produced an empty Chroma collection because the plugin always called `lookup.space()`, which 404'd for KB-backed BoKs and aborted the pipeline before orphan cleanup could run. The empty collections then served RAG queries, producing hallucinated answers. This PR routes those events to `lookup.knowledgeBase()` so they ingest correctly.

## Relationship to PR #98

PR #98 contains only the fix commit; this PR is `#98 + spec docs`. Recommended path: close #98 in favour of this combined PR so the spec lands together with the code (matches the PR #91 pattern for spec 030).

## Test plan

- [x] `poetry run pytest tests/plugins/test_ingest_space.py -q` — 37 passed (12 new).
- [x] `poetry run pytest --ignore=tests/evaluation -q` — full non-evaluation suite green.
- [x] `poetry run ruff check core/ plugins/ tests/` — clean.
- [x] `poetry run pyright core/ plugins/` — no new errors.
- [ ] After deploy to acceptance: trigger ingest for a known KB-backed VC (e.g. `test-vc-flow-003`, BoK `e2cf604d-…`); confirm pod logs show the KB reader was used and the resulting collection is populated. Re-ask the previously-hallucinated question; expect a grounded answer.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Support ingesting knowledge-base–backed bodies alongside space content; KB roots are tagged as "knowledge".
  * Dispatcher routes ingestion by body-of-knowledge type with safe fallback to space ingestion.
  * INFO logging now reports BoK id, type, and purpose during ingestion.

* **Documentation**
  * Added spec, plan, checklist, quickstart, research, data-model, and contract docs for KB routing.

* **Tests**
  * Added tests covering KB reader, dispatcher, and plugin routing/edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/alkem-io/virtual-contributor/pull/100)

### CODE REVIEW

File: tests/plugins/test_ingest_space.py [hunk 1]
```diff
@@ -689,3 +696,167 @@ async def side_effect(url):
...
+            new=AsyncMock(return_value=[]),
+        ) as kb_mock, patch(
+            "plugins.ingest_space.space_reader.read_space_tree",
+            new=AsyncMock(return_value=[]),
+        ) as space_mock:
+            result = await plugin.handle(event)
+
+        assert result.result == "success"
+        space_mock.assert_awaited_once()
+        kb_mock.assert_not_awaited()
```
[the remark this answers is not in the document]
Comment (valentinyanakiev) [reply tests/plugins/test_ingest_space.py:872]: Addressed in 915384c. The two `TestIngestSpacePluginDispatchesOnType` cases now assert, in addition to the routing check, that no LLM/embeddings work was performed on the zero-document cleanup branch (`llm.calls`, `embeddings.calls`, `embeddings.query_calls` all empty) and that no collection deletion was issued (`store.deleted` empty). On the nitpick about patching `read_body_of_knowledge` directly: declined. Patching the dispatcher would weaken the test — it would only prove that `event.type` reached `read_body_of_knowledge`, not that routing maps that value to the correct leaf reader. The current leaf-level patches give the stronger end-to-end guarantee, and the dispatcher itself is covered separately in `TestBodyOfKnowledgeDispatcher`.

**Comment by valentinyanakiev** [comment]
CodeRabbit review triage (commit 915384c):

**Outside-diff comment — `plugins/ingest_space/plugin.py` log messages "Space …" → "BoK …":** addressed. Both the zero-document cleanup info log and the catch-all exception log were updated to read "BoK …" since this plugin now serves both `alkemio-space` and `alkemio-knowledge-base` types.

**Inline comment — assert LLM/knowledge-store interactions in plugin dispatch tests:** addressed. See thread for detail.

**Nitpick — patch `read_body_of_knowledge` dispatcher directly:** declined. Patching the dispatcher would prove only that `event.type` reached `read_body_of_knowledge`; it would not prove the routing maps that value to the correct leaf reader. The current leaf-level patches give a stronger end-to-end guarantee, and the dispatcher itself is covered by `TestBodyOfKnowledgeDispatcher`.

Gates after fixup: 528 pytest pass, ruff clean, pyright 0 errors.
